jacksono/flask-realworld-example-app
# Enable articles to be placed into categories

### Background context for this PR
Apart from using tags we can also use categories to group articles. Categories are hierarchical so a category can have a parent and a category can have children and sub-children. We can then be able to list articles that belong to a category (or a categories in the same hierarchy). 

#### What does this PR do?
* Add a `Category` table which is self referencing to enable creating of a hierarchical structure
* Adds foreign key referencing between the `Article` table and the `Category` table to enable articles to be added to a category.
* Adds API endpoints to list, add, edit and delete categories
* Adds unit tests for new endpoints and the new feature

#### How should PR reviewers test this?
(_check the readme to see how to set up the project on your computer, dont forget to run migrations_)
- Unit tests:
  - Run the unit tests using `flask test` on the terminal and ensure that all the tests pass
- Manual tests:
  - Use the postman collection found at the project root directory to try out the end point and watch test for any errors/bugs.

##### TODOs (for code/project improvement)
* [ ] Handle more edge cases
* [ ] Refactor the Category controller and tests into their own modules
